src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -606,7 +607,14 @@ TaskDirectoryName(uint64 jobId, uint32 taskId)
 }
 
 
-/* Constructs a standardized partition file path for given directory and id. */
+/*
+ * PartitionFilename returns a partition file path for given directory and id
```

Comment:
I cannot follow the difference between `PartitionFilename ()` and `UserPartitionFilename ()` 😞 

My main question is that, why do we still need `PartitionFilename ()` ?

Reply:
PartitionFilename:
`base/pgsql_job_cache/job_201010/task_101106/p_00000`

UserPartitionfilename:
`base/pgsql_job_cache/job_201010/task_101106/p_00000.marco`

`PartitionFilename` is called to contruct a transmit statement for a partition file in `worker_fetch_partition_file`. On the wire, the paths do not have the username suffixes. The suffix is only added before reading or writing a file.

Reply:
Ok, that makes more sense now. Thanks!